include/boost/math/special_functions/prime_sieve.hpp

```diff
@@ -14,9 +14,9 @@
 #include <vector>
 #include <iterator>
 #include <cmath>
-
-#if __has_include(<execution>)
 #include <thread>
+
+#if __cplusplus >= 201703
```

Comment:
I admit to having preferred the `__has_include` since that would protect against compilers with partial C++17 support. Maybe both?

Reply:
@NAThompson It looks like every compiler that does not support `<execution>` also do not support [`__has_include`](https://en.cppreference.com/w/cpp/compiler_support)

Reply:
I don't think that's still _quite_ right: the outlier is MSVC which still does not define __cpluplus to a sensible value!

ordinarily the right way to do this is:

```
#ifdef __has_include
#if __has_include(<execution>)
#include <execution>
#define BOOST_MATH_USE_STD_EXECUTION_POLICIES // if required?
#endif
#endif
```

But.... sadly that still fails if the compilation mode is not C++17 and <execution> happens to be in the include path :(  So I think we need to add a Boost.Config macro (feel free to nag me about that!).

Reply:
@jzmaddock What about:

```
#ifdef _MSVC_LANG
#if _MSVC_LANG >= 201703 // _MSVC_LANG == __cplusplus: https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
#include <execution>
#endif
#else
#if __cplusplus >= 201703
#include <execution>
#endif
#endif
```
That compiles fine with `-std=c++11` and should be portable